src/RouteTransition.jsx

```diff
@@ -51,20 +35,22 @@ const RouteTransition = React.createClass({
     }
 
     // TODO: maybe access route path from children for pathname?
-    return [{
-      key: this.props.pathname,
-      data: this.props.children,
-      style: ensureSpring(this.props.atActive)
-    }];
-  },
+    return [
+      {
+        key: this.props.pathname,
+        data: this.props.children,
+        style: ensureSpring(this.props.atActive)
+      }
+    ];
+  }
 
   willEnter() {
-    return this.props.atEnter;
-  },
+    return () => this.props.atEnter;
```

Comment:
i would prefer we don't add extra functions and just bind these reference in the constructor. can you update that?